### PR TITLE
electron-cash: update and fix build

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

Update electron-cash and add patch from pull request to remove qdarkstyle version pin

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/electron-cash/default.nix [hunk 1]
```diff
@@ -3,15 +3,23 @@
...
     repo = "Electron-Cash";
     rev = version;
-    sha256 = "sha256-ALIrNnhpX46xdQdfJdx/9e/QtdyBEgi5xLrbuOBJR7o=";
+    sha256 = "sha256-/5b9V1m7hBI6qsLr1/nE9bWj2SPX9SmdlzzFqS3Nc6Q=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "relax-qdarkstyle-version-bound.patch";
+      url = "https://github.com/Electron-Cash/Electron-Cash/pull/2396.patch";
```
Comment (oxalica) [inline pkgs/applications/misc/electron-cash/default.nix:18]: Please don't use the PR diff URL. They are not deterministic and may cause build failure later. Use commit diff URL instead:
- `https://github.com/Electron-Cash/Electron-Cash/commit/086c7f797f60e9f7c84bc622d867c98cb3b743ee.patch`
- `https://github.com/Electron-Cash/Electron-Cash/commit/8d0aa028e1665196e6c8643eae1fbf9fae33e9f4.patch`

Also, let's add a comment for the PR URL before `fetchpatch`.

**Review by oxalica — CHANGES_REQUESTED** [review]
The PR title should also be like `electron-cash: update and fix build`.

**Comment by oxalica** [comment]
Is this PR still active?

**Comment by Lassulus** [comment]
superseded by https://github.com/NixOS/nixpkgs/pull/165123

**Pull request closed without merging**